packages/gatsby-source-preview/gatsby-node.js

```diff
@@ -93,7 +135,14 @@ exports.sourceNodes = (
         }
       }
 
-      processDirectoryTree(tree)
+      let tree
+      if (changedFile) {
+        // tree = { path: changedFile }
+        changedFile.forEach(elem => processDirectoryTree({ path: elem }))
```

Comment:
I was asking myself if, for a changed file only, we could not instead use `getnode()` to get all files contents that did not change.